kafka-rest/src/main/java/io/confluent/kafkarest/controllers/PartitionManagerImpl.java

```diff
@@ -73,7 +79,23 @@ public CompletableFuture<Optional<Partition>> getPartition(
       String clusterId, String topicName, int partitionId) {
     return topicManager
         .getTopic(clusterId, topicName)
-        .thenApply(topic -> checkEntityExists(topic, "Topic %s cannot be found.", topic))
+        .handle(
+            (value, exception) -> {
+              if (exception != null) {
+                log.error("Caught exception while get topic from cluster", exception);
+                if (exception.getCause() instanceof UnknownTopicOrPartitionException) {
+                  String exceptionMessage =
+                      String.format(
+                          "This server does not host this topic-partition for topic %s", topicName);
+                  throw new UnknownTopicOrPartitionException(exceptionMessage, exception);
+                } else if (exception instanceof NotFoundException
+                    || exception.getCause() instanceof NotFoundException) {
+                  throw new NotFoundException(exception);
+                }
+                throw new RuntimeException(exception);
```

Comment:
Again, why do we need the extra layer of exception here?

Reply:
The type of exception here is Throwable, we can't re-throw it without handle it.
<img width="653" alt="Screen Shot 2022-09-08 at 10 45 53 PM" src="https://user-images.githubusercontent.com/7269552/189280009-b1c125be-f2f6-4232-b9ec-35a91b4bbe37.png">